standardization/src/main/scala/za/co/absa/enceladus/standardization/interpreter/stages/TypeParser.scala

```diff
@@ -85,17 +83,30 @@ object TypeParser {
                     path: String,
                     origSchema: StructType,
                     parent: Option[Parent] = None): TypeParser = {
-    val parserClass = field.dataType match {
-      case _: ArrayType     => ArrayParser
-      case _: StructType    => StructParser
-      case _: NumericType   => NumericParser
-      case _: StringType    => StringParser
-      case _: BooleanType   => BooleanParser
-      case _: DateType      => DateParser
-      case _: TimestampType => TimestampParser
+    field.dataType match {
+      case _: ArrayType                 => ArrayParser(field, path, origSchema, parent)
+      case _: StructType                => StructParser(field, path, origSchema, parent)
+      case _: ByteType                  => IntegralParser(field,
+        path,
+        origSchema,
+        parent,
+        Set(ShortType, IntegerType, LongType),
+        Byte.MinValue,
+        Byte.MaxValue)
+      case _: ShortType                 =>
+        IntegralParser(field, path, origSchema, parent, Set(IntegerType, LongType), Short.MinValue, Short.MaxValue)
+      case _: IntegerType               =>
+        IntegralParser(field, path, origSchema, parent, Set(LongType), Int.MinValue, Int.MaxValue)
+      case  _: LongType                 =>
+        IntegralParser(field, path, origSchema, parent, Set.empty, Long.MinValue, Long.MaxValue)
+      case _: FloatType | _: DoubleType => FractionalParser(field, path, origSchema, parent)
+      case _: DecimalType               => DecimalParser(field, path, origSchema, parent)
+      case _: StringType                => StringParser(field, path, origSchema, parent)
+      case _: BooleanType               => BooleanParser(field, path, origSchema, parent)
+      case _: DateType                  => DateParser(field, path, origSchema, parent)
+      case _: TimestampType             => TimestampParser(field, path, origSchema, parent)
```

Comment:
If I can have a full on a suggestion about keeping the repeating part minimal. I liked the previous approach a lot. The styling from the example below isn't perfect but should showcase what I mean and I know this is not a full solution, but I hope it is enough to illustrate.

```scala
val parserClass = field.dataType match {
      case _: ArrayType     => ArrayParser
      case _: StructType    => StructParser
      case _: ByteType      => IntegralParser(Set(ShortType, IntegerType, LongType), Byte.MinValue, Byte.MaxValue)
      case _: ShortType     => IntegralParser(Set(IntegerType, LongType), Short.MinValue, Short.MaxValue)
      case _: IntegerType   => IntegralParser(Set(LongType), Int.MinValue, Int.MaxValue)
      case  _: LongType     => IntegralParser(Set.empty, Long.MinValue, Long.MaxValue)
      case _: FloatType |
           _: DoubleType    => FractionalParser
      case _: DecimalType   => DecimalParser
      case _: StringType    => StringParser
      case _: BooleanType   => BooleanParser
      case _: DateType      => DateParser
      case _: TimestampType => TimestampParser
      case t =>
        throw new IllegalStateException(s"${t.typeName} is not a supported type in this version of Enceladus")
    }
    parserClass(field, path, origSchema, parent)
```
and IntegralParser

```scala
private final case class IntegralParser(field: StructField,
                                          path: String,
                                          origSchema: StructType,
                                          parent: Option[Parent]
                                         )(overflowableTypes: Set[DataType ],
                                          minValue: Long,
                                          maxValue: Long) extends NumericParser
```

Reply:
I find this to be more the functional way and there is less white noise and fewer places for refactoring